### ISSUE TITLE
[Question] onManaChange(creature, attacker, manaChange, origin)

### ISSUE THREAD

**Issue by conde2** [otland/forgottenserver#1700]
How do I know if the player is gaining mana or losing mana ?

Because manaChange is always positive.

**Comment by conde2** [comment]
https://github.com/otland/forgottenserver/blob/70590833b25f6b6a137c5e45ce29b254edddadf8/src/game.cpp#L4105


Shouldn't it be manaLoss?
https://github.com/otland/forgottenserver/blob/70590833b25f6b6a137c5e45ce29b254edddadf8/src/game.cpp#L4120

**Comment by conde2** [comment]
@WibbenZ @Mkalo
Could you guys check this issue please ?

**Comment by ranisalt** [comment]
> Shouldn't it be manaLoss?

No. `manaLoss` is the absolute value (on the else branch, `manaChange` is non-positive so `manaLoss` is non-negative). The value sent to `onManaChange` can be positive or negative, from the code you shown.

**Comment by Mkalo** [comment]
I do have this:
https://gist.github.com/Mkalo/2b1c4bcd772e161a1200cfeb820d3aba

But I never created a PR cause I didn't have time before, I could do it if this is okay.

@ranisalt

**Comment by WibbenZ** [comment]
Imo this issue is kinda related to #1211, seeing as the functions need an overhaul.

**Comment by Mkalo** [comment]
I don't know if onManaChange should be executed for spell casting anyway, (It doesnt use any combat function just Creature::changeMana).

We could change that and make it fire a combat (Game::combatChangeMana) and use ORIGIN_SPELL.

**Comment by WibbenZ** [comment]
As mark wrote this function should be used as much as we can.
Look at #1226
